src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -709,12 +712,13 @@ public function addProductAction(int $orderId, Request $request): Response
         /** @var OrderForViewing $orderForViewing */
         $orderForViewing = $this->getQueryBus()->handle(new GetOrderForViewing($orderId));
 
-        $updatedProducts = $orderForViewing->getProducts()->getProducts();
-        $newProducts = [];
-        for ($i = count($previousProducts); $i < count($updatedProducts); ++$i) {
-            $newProducts[] = $updatedProducts[$i];
+        $updatedProducts = [];
+        foreach ($orderForViewing->getProducts()->getProducts() as $orderProductForViewing) {
+            $updatedProducts[$orderProductForViewing->getOrderDetailId()] = $orderProductForViewing;
         }
 
+        $newProducts = array_diff_key($updatedProducts, $previousProducts);
```

Comment:
Nice catch, even though it shouldn't be useful any more since you refresh the whole list